pkg/certificates/certificates_test.go

```diff
@@ -45,6 +47,27 @@ var (
 	}
 )
 
+type noListener struct {
```

Comment:
It seems it's getting messier than I expected.

Reply:
Because it added noListener and sync primitive Event (which would have also needed few tests) to support one test-case. The noListener had a stub address.